fix: improve git handler regexes

sportshead opened the pull request:
* Uses `git remote set-url --push origin` to get the url instead of parsing from `git remote -v`
* Clean up regex matching code, use `string.match` instead of `string.find`
* Allow optional .git for ssh remotes
* Detect owner in owner#issue format
* Detect owner, repo in owner/repo#issue format
* Add `handler_options.git_remotes` that takes a `string[]` or `function(fname: string): string[]`, indicating the order of remote names to search for a valid URL

Fixes #45, #56
Supersedes #24

phanen commented:
Would be better to use `git remote set-url --no-push`? 
Sometimes I just pull from one repo and push to my fork repo, so the `/issues` link may not exist in my own repo.

chrishrb commented:
for me this handler works as expected. 👍 

@sportshead what do you think about the suggestion from @phanen? After this is resolved, this PR can be merged.

sportshead commented:
> for me this handler works as expected. 👍
> 
> @sportshead what do you think about the suggestion from @phanen? After this is resolved, this PR can be merged.

@chrishrb @phanen - thanks for the comments, I'll add an option to allow configuring whether to use the push url or not, to give the user the most flexibility.

sportshead commented:
Apologies for the late fixes @chrishrb. Got very busy and forgot about this PR. I've rebased the PR and added a `git_remote_push` option defaulting to false, which should resolve @phanen's suggestion.
